src/vma/lwip/tcp_out.c

```diff
@@ -1121,6 +1121,11 @@ tcp_output(struct tcp_pcb *pcb)
        #if TCP_OVERSIZE_DBGCHECK
          seg->oversize_left = 0;
        #endif /* TCP_OVERSIZE_DBGCHECK */
+       if ((pcb->in_flag_hndlg & TCP_SYN) && (get_tcp_state(pcb) & SYN_RCVD)) {
+         pcb->in_flag_hndlg = 0;
+         pcb->rcv_nxt = 1;
+         seg->tcphdr->seqno = htonl(htonl(seg->tcphdr->seqno)-1);
+       }
```

Comment:
This will probably affect the fast path, can we avoid this addition ?

Reply:
This functionality probably can be moved to the slowtmr/fasttmr? I will review it.